compiler/modules/front/expr.bal

```diff
@@ -36,6 +36,9 @@ type Binding record {|
     // In the case of the nested narrowing, this points all the way
     // back to the original explicit binding, not to the previous narrowing.
     Binding? unnarrowed = ();
+    Position pos;
+    // If this binding is an assignment, reg number of the invalidated NarrowRegister
+    int? invalidated = ();
```

Comment:
I think we need a longer name for this.

What distinguishes a Binding that represents an assignment? Is it invalidated being non-nil? If so, we should put `assignment` in the name.

Why is this a register number rather than a register?

Reply:
I changed to `NarrowRegister? assignmentInvalidateReg`. Now with the type being `NarrowRegister` I think even `invalidatedReg` makes sense. wdut?